enhancements/network/egress-ip-per-destinations.md

```diff
@@ -0,0 +1,327 @@
+CRs can consume.
+
+If greater than one `EgressIPTraffic` CRs are selected, the networks are added together, and if a destination IP is within one
+of the networks, then the pods source IP will be the `EgressIP`.
+If `TrafficSelector` is not specific, existing `EgressIP` behaviour is conserved.
+
+### Workflow Description
+
+**cluster admin** creates a set of namespaces for tenants to use.
+Tenant has requested that its applications need a consistent source IP
+for 2 different traffic paths - one health signal traffic and one for work load traffic.
+The cluster admin updates a dedicated egress node by attaching two new networks on separate interfaces.
+The cluster admin labels this node egress-able.
+The cluster admin creates two `EgressIPTraffic` CRs, specifies a destination network in each and labels each CR differently.
+The cluster admin creates two `EgressIP` CRs each containing an IP that falls within the previously added networks range.
+The two `EgressIP`s selects the same set of pods but different `TrafficSelector`s to select the previously created `EgressIPTraffic` CRs.
+The `EgressIP`s are automatically assigned to the secondary host interfaces on the egress node.
+
+The cluster admin gives permissions to the **application admin** to the namespaces.
+
+The **application admin** deploys its application and the application has a consistent but different source IP for the
+health signal traffic and the work load traffic.
+
+### API Extensions
+
+Add `TrafficSelector` field to `EgressIPSpec`:
+
+```golang
+// EgressIPSpec is a desired state description of EgressIP.
+type EgressIPSpec struct {
+	// EgressIPs is the list of egress IP addresses requested. Can be IPv4 and/or IPv6.
+	// This field is mandatory.
+	EgressIPs []string `json:"egressIPs"`
+	// TrafficSelector applies the egress IP only to the network traffic defined within the selected EgressIPTraffic(s).
+	// If not set, all egress traffic is selected.
+	// +optional
+	TrafficSelector metav1.LabelSelector `json:"trafficSelector"`
+	// NamespaceSelector applies the egress IP only to the namespace(s) whose label
+	// matches this definition. This field is mandatory.
+	NamespaceSelector metav1.LabelSelector `json:"namespaceSelector"`
+	// PodSelector applies the egress IP only to the pods whose label
+	// matches this definition. This field is optional, and in case it is not set:
+	// results in the egress IP being applied to all pods in the namespace(s)
+	// matched by the NamespaceSelector. In case it is set: is intersected with
+	// the NamespaceSelector, thus applying the egress IP to the pods
+	// (in the namespace(s) already matched by the NamespaceSelector) which
+	// match this pod selector.
+	// +optional
+	PodSelector metav1.LabelSelector `json:"podSelector,omitempty"`
+}
+```
+
+New CRD:
+
+```golang
+// +genclient
+// +genclient:nonNamespaced
+// +kubebuilder:object:root=true
+// +kubebuilder:subresource:nostatus
+// +kubebuilder:resource:shortName=eipd,scope=Cluster
```

Comment:
Why `eipd`? `eipt` for `EgressIPTraffic` would be more fitting, no?

Reply:
typo fixed :)